### PR TITLE
Add support for SASL EXTERNAL

### PR DESCRIPTION
**Pull request by progval**
Closes GH-246.

Tested on Ergo.Chat.

This might be a little confusing, as it requires a `method` configured in the `user` section of the config, and the `certificate` configured in the `server` section. (Though it works with only the latter on some servers, but this is called CertFP, not SASL EXTERNAL.)

I was not able to do otherwise without refactoring all the configuration handling.

Additionally, the reason for adding this new `server->certificate` field is that this is a TLS cert, not an ECDSA cert like the existing `user->certificate` field. (btw, [ECDSA authentication is deprecated by IRCv3 (including the original author)](https://github.com/ircv3/ircv3.github.io/pull/11))

### CODE REVIEW

**Comment by progval** [comment]
> however the GUI "login method" dropdown doesn't seem to be updated yet.

Indeed, I forgot. Done.

>  Also, I cannot join channels

Yeah, doesn't seem related

**Comment by ljrk0** [comment]
Great work, thanks for digging into this!

File: data/ui/connect_panel.glade [hunk 1]
```diff
@@ -580,6 +621,7 @@
       <widget name="login_password_entry"/>
       <widget name="remember_login_password_check_button"/>
       <widget name="login_cert_file_chooser_button"/>
+      <widget name="client_cert_file_chooser_button"/>
```
Comment (SilverRainZ) [inline data/ui/connect_panel.glade:624]: The  client_cert name is coufusing, how about rename the login_cert to ecdsa_cert and this to another better name? such as sasl_external_cert?

File: src/sirc/sirc.c [hunk 2]
```diff
@@ -337,6 +337,19 @@ static void on_connect_ready(GObject *obj, GAsyncResult *res, gpointer user_data
...
+         if (cert_filename) {
+             g_autoptr(GTlsCertificate) cert = g_tls_certificate_new_from_file(cert_filename, &err);
+             if (err){
+                 on_connect_fail(sirc, err->message);
+                 g_error_free(err);
+                 return;
+             }
+             g_tls_connection_set_certificate(G_TLS_CONNECTION(tls_conn), cert);
+            LOG_FR("Using client TLS certificate");
+         }
```
Comment (SilverRainZ) [inline src/sirc/sirc.c:351]: I thinks this is why we have to configure cert in server section? Keep status quo is fine. We can expose a SIRC_EVNET in future.

File: data/ui/connect_panel.glade (hunk 1 above)
Comment (SilverRainZ) [reply data/ui/connect_panel.glade:624]: Yes, we can add a new confval like `login.ecdsa_certificate`, but keep it compatible with the old `login.certificate` confval.

**Pull request merged**